File: js/src/webServer/webServer.js
var http = require('http');
var path = require('path');
var fs = require('fs');

function handleRequest(request, response){
  function handleError() {
    response.statusCode = 404;
    response.end();
  }

  try {
    var index = request.url.substr(1);
  } catch (_) {
    return handleError();
  }

  var filePath = path.join(
      process.env.RESOURCES_PATH,
      "best-selling-2015-" + index + '.json'
  );

  fs.exists(filePath, function (exists) {
    if (!exists) {
      return handleError();
    }

    fs.readFile(filePath, 'utf8', function (error, contents) {
      if (error) {
        return handleError();
      }

      response.writeHead(200, {'Content-Type': 'application/json'});
      response.end(contents);
    });
  });
}

function main() {
  var server = http.createServer(handleRequest);
  server.listen(8080);
}
main();

File: go/src/webServer/webServer.go
package main

import "fmt"
import "net/http"
import "os"
import "path"
import "io/ioutil"

func handler(w http.ResponseWriter, r *http.Request) {
	index := r.URL.Path[1:]
	filePath := path.Join(
		os.Getenv("RESOURCES_PATH"),
		"best-selling-2015-" + string(index) + ".json",
	)

	contents, err := ioutil.ReadFile(filePath)
	if err != nil {
		w.WriteHeader(http.StatusNotFound)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	fmt.Fprint(w, string(contents))
}

func main() {
	http.HandleFunc("/", handler)
	http.ListenAndServe(":8080", nil)
}